pkg/expr/transform.go

```diff
@@ -56,7 +56,7 @@ type TimeRange struct {
 
 // TransformData takes Queries which are either expressions nodes
 // or are datasource requests.
-func (s *Service) TransformData(ctx context.Context, req *Request) (r *backend.QueryDataResponse, err error) {
+func (s *Service) TransformData(ctx context.Context, req *Request) (r *Response, err error) {
```

Comment:
I believe we should keep these the same type so there are not blockers to extracting the service in the future.

So I would rather see one of the two ugly choices:
 - A) Attach it the the first frame (preferred)
 - B) Attach it to all frames

And then look for it somewhere outside the service (like the FE if that is possible I think would be best)

This also means when there is no data, the warning will be omitted, which is also unfortunate but okay I think. In someday-future, I have a design where an empty response always include at least one frame (with length 0/nil) to indicate an empty response.  So once that contract is in place that will be solvable. (Or, as long as alerting understands it correctly as NoData, SSE could start to implement this now on output. There may be some FE concerns outside of alerting here but I don't think this is normalized yet anyways.)

Reply:
As an option, we go with option B) and then ask FE to dedup messages. In that case if there is the same notice in all frames, FE will squash it and show just a single warning. In the case of different notices, FE will show them all.

Reply:
That works as well as long as we keep the information needed in `[]data.Frame`, (well `[]*data.Frame`, but that is a mistake that would be nice to get to one day as far as I can tell).